tests/js/legendTests.js

```diff
@@ -0,0 +1,299 @@
+                args: ["{that}.dom.row", "mouseover", "gpii.tests.chartAuthoring.mouseOverListener"]
+            }
+        },
+        members: {
+            mouseOverListenerCalled: false
+        }
+    });
+
+    gpii.tests.chartAuthoring.objectArray = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayAdd = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id4",
+        value: 26,
+        label: "Five"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayRemove = [{
+        id: "id0",
+        value: 15,
+        label: "One"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayChangeInPlace = [{
+        id: "id0",
+        value: 36,
+        label: "I"
+    }, {
+        id: "id1",
+        value: 67,
+        label: "II"
+    }, {
+        id: "id2",
+        value: 26,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.objectArraySorted = [{
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id0",
+        value: 15,
+        label: "One"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayAddSorted = [{
+        id: "id1",
+        value: 67,
+        label: "Two"
+    }, {
+        id: "id3",
+        value: 45,
+        label: "Four"
+    },
+    {
+        id: "id4",
+        value: 26,
+        label: "Five"
+    },
+    {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id0",
+        value: 15,
+        label: "One"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayRemoveSorted = [{
+        id: "id1",
+        value: 67,
+        label: "Two"
+    },
+    {
+        id: "id2",
+        value: 20,
+        label: "Three"
+    }, {
+        id: "id0",
+        value: 15,
+        label: "One"
+    }];
+
+    gpii.tests.chartAuthoring.objectArrayChangeInPlaceSorted = [{
+        id: "id1",
+        value: 67,
+        label: "II"
+    }, {
+        id: "id0",
+        value: 36,
+        label: "I"
+    }, {
+        id: "id2",
+        value: 26,
+        label: "Three"
+    }];
+
+    gpii.tests.chartAuthoring.mouseOverListener = function (data, i, that) {
+        that.mouseOverListenerCalled = true;
```

Comment:
This test is to verify the provided API successfully hooks up D3 listeners. It doesn't care how many times the listener is called. In fact, the listener is only called once in the test. If we track the number of times the listener gets called, if by any chance when the test runs and the mouse is hovering over the tested container (the container for drawing is visible), the tracked number might be 2 instead of 1, which will cause the test to fail. This failure doesn't indicate the listener isn't attached properly.